components/style/properties/longhand/inherited_text.mako.rs

```diff
@@ -936,6 +936,111 @@ ${helpers.single_keyword("text-align-last",
     }
 </%helpers:longhand>
 
+<%helpers:longhand name="text-emphasis-position" animatable="False" products="none">
+    use std::fmt;
+    use values::computed::ComputedValueAsSpecified;
+    use values::NoViewportPercentage;
+    use style_traits::ToCss;
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum VerticalValue {
+        Right,
+        Left
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum HorizontalValue {
+        Over,
+        Under
+    }
+
+    #[derive(Debug, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum SpecifiedValue {
+        Value(HorizontalValue, VerticalValue)
+    }
+
+    pub mod computed_value {
+        pub type T = super::SpecifiedValue;
+    }
+
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+    impl NoViewportPercentage for SpecifiedValue {}
+
+    pub fn get_initial_value() -> computed_value::T {
+        SpecifiedValue::Value(HorizontalValue::Over, VerticalValue::Right)
+    }
+
+    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
```

Comment:
I might be misunderstanding `Either`, but if I wanted to use it, then instead of `VerticalValue` and `HorizontalValue` wouldn't I need to make types for all the variations that `right`, `left`, `over`, and `under` could take? If I make it like above, then wouldn't I only be able to return a `HorizontalValue` or a `VerticalValue`? I need to return one of both.